src/gui/shellwidget/shellwidget.cpp

```diff
@@ -239,13 +239,13 @@ void ShellWidget::paintUnderline(
 static QPainterPath GetUndercurlPath(QRect cellRect) noexcept
 {
 	const QLine underline{ GetUnderline(cellRect) };
-	const QPoint& start{ underline.p1() };
-	const QPoint& end{ underline.p2() };
+	const QPointF& start{ underline.p1() };
+	const QPointF& end{ underline.p2() };
 
-	QPainterPath path(start);
-	static constexpr int offset[8]{ 1, 0, 0, 1, 1, 2, 2, 2 };
-	for (int i = start.x() + 1; i <= end.x(); i++) {
-		path.lineTo(QPoint(i, start.y() - offset[i%8]));
+	QPainterPath path{};
```

Comment:
Let's use the old CTOR. We can still fix this with `for (int i = start.x() + 1`.

Qt is vague on how `toLine` behaves without an explicit `currentPoint`, which was previously provided in the CTOR.